smarts/core/sumo_road_network.py

```diff
@@ -269,6 +274,60 @@ def _snap_internal_edges(self, lane_to_poly, snap_threshold=2):
                 lane_shape = Polygon(snap(lane_shape, outgoing_shape, snap_threshold))
                 lane_to_poly[lane_id] = lane_shape
 
+    def _snap_internal_holes(self, lane_to_poly, snap_threshold=2):
+        for lane_id in lane_to_poly:
+            lane = self.lane_by_id(lane_id)
+
+            # Only do snapping for internal edge lane holes
+            if not lane.getEdge().isSpecial():
+                continue
+            lane_shape = lane_to_poly[lane_id]
+            for x, y in lane_shape.exterior.coords:
+                for nl, dist in self.nearest_lanes(
+                    (x, y),
+                    max(10, 2 * self._default_lane_width),
+                    include_junctions=False,
+                ):
+                    if nl:
+                        nl_shape = lane_to_poly.get(nl.getID())
+                        if nl_shape:
+                            lane_shape = Polygon(
+                                snap(lane_shape, nl_shape, snap_threshold)
+                            )
```

Comment:
I could suggest using continue here to help alleviate the tabbing.

```suggestion
                    if not nl:
                        continue
                    nl_shape = lane_to_poly.get(nl.getID())
                    if nl_shape:
                        lane_shape = Polygon(
                            snap(lane_shape, nl_shape, snap_threshold)
                        )
```